hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java

```diff
@@ -700,6 +701,9 @@ public void bootstrapOzoneManager(String omNodeId) throws Exception {
     while (true) {
       try {
         basePort = 10000 + RANDOM.nextInt(1000) * 4;
+        if (!isPortAvailable(basePort)) {
```

Comment:
Rather than trying incremental ports, can we just ask the OS for a free one, eg:

private int getFreePort() {
  ServerSocket ss = null;
  try {
    ss = new ServerSocket(0);
    return ss.getLocalPort();
  } catch { 
     ...
  } finally {
    if (ss != null) {
      ss.close();
    }
}

Reply:
The discussion here suggests a few problems etc with various approaches:

https://stackoverflow.com/questions/434718/sockets-discover-port-availability-using-java

Reply:
@sodonnel Thanks for review.
Yeah, I was checking the same page, too.

Normally we don't use ports under 1000, but in docker it should be fine?
Another case is if the port returned is larger than 65531, it will exceed the port range?

Reply:
There are already some tests that use random port using this `ServerSocket(0)` technique.  We could just call this method to get OM port:

https://github.com/apache/ozone/blob/d96399142f1f84f54d6dd783dbff6b8415f1c1e6/hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/SCMTestUtils.java#L115-L122

Reply:
Sure. 
Do we need to call the method 4 times, or add a restriction that basePort is under 65532?

Reply:
I'd go for calling it once for each distinct port.  Any port can be occupied independently.

Reply:
I wonder if the method could give the same port back on consecutive calls. I guess we can try it and see.

It would be great if we could just pass port zero to OM and let it start on a free port by itself, but I guess other things need to connect to it, so they need the port, which would be unknown. Therefore the approach we are taking here is likely the most sensible.